Adds two new basis functions for creating AltAz masks.

tribeiro opened the pull request:
- HADecAltAzPatchBasisFunction: Create masks using a combination of patches defined as hour angle, declination, altitude and azimuth limits.
	- MeridianStripeBasisFunction: Create masks using a combination of meridian mask and zenith pad. Also allows different weights to be applied to different regions.

yoachim commented on python/lsst/sims/featureScheduler/basis_functions.py:
```diff
@@ -216,6 +216,162 @@ def __call__(self, indx=None):
+            else:
+                ha_mask = np.ones(hp.nside2npix(self.nside), dtype=bool)
+
+            if 'dec_min' in patch and 'dec_max' in patch:
```
Since the dec map doesn't change, you could move the `dec_mask` calculation to the __init__ and do it just once. You could add an RA mask to the __init__ as well since it's the lone coordinate left out.

tribeiro commented:
Those are great ideas. As suggested, I moved the declination mask to `__init__` on `HADecAltAzPatchBasisFunction` and added unit tests for both basis functions. The tests are quite simple, just make sure the default masks generates some observable region. In the future we can think about adding more complex tests.
